pkg/aggregator/aggregator_test.go

```diff
@@ -575,9 +581,12 @@ func TestTags(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			defer pkgconfigsetup.Datadog().SetWithoutSource("basic_telemetry_add_container_tags", nil)
-			pkgconfigsetup.Datadog().SetWithoutSource("basic_telemetry_add_container_tags", tt.tlmContainerTagsEnabled)
-			agg := NewBufferedAggregator(nil, nil, tt.hostname, time.Second)
+			mockConfig := configmock.New(t)
+			mockConfig.SetWithoutSource("basic_telemetry_add_container_tags", tt.tlmContainerTagsEnabled)
```

Comment:
`mockConfig` created and not used? Are these changes necessary for the tagger replacement? If not, would it be possible to do that in a separate PR?

Reply:
I'm not sure I follow. `basic_telemetry_add_container_tags` is used by the aggregator, not the tagger, and is read from the `pkgconfigsetup`. If it is set on the mock, will the code under testing here see the correct value?

Reply:
Prior to this change, we were not using `fxutil.Test` in the tests. Now, that we do and it happens to be that we use the mock confog as part of the tagger mock https://github.com/DataDog/datadog-agent/blob/32e9c48ca650d663fc94c311167697cc91e256ab/comp/core/tagger/taggerimpl/tagger_mock.go#L70. 

The config mock, set the global `pkgconfigsetup.Datadog()` as part of the initialization as part of the initialization process https://github.com/DataDog/datadog-agent/blob/c260596a4c871702ab626fda46adb318e140b780/pkg/config/mock/mock.go#L52

If we do not change the test to use the `mockConfig := configmock.New(t)` the code inside that `NewBufferedAggregator` that fetches the global config would not use the right config object.

Reply:
So both the mock module and this code create and install a global config. And this one wins, because it happens last. I guess this works, but is rather unexpected. Would it be possible to have one mock config in the test?